_posts/2023-02-13-semantic-search-solutions.md

```diff
@@ -0,0 +1,106 @@
+
+If you’re interested in semantic search performance, look for detailed benchmarking documentation that we’ll be releasing. This documentation will include such benchmarking data as query latency, ingestion latency, and query throughput.
+
+### Option 2: Semantic search with a tuned model
+
+To build a custom solution, ideally, you need a dataset that consists of `(query, relevant passage)` pairs from the chosen domain to train a model so that it performs well on that domain. In the OpenSearch context, it is common to have passages, but not queries. The synthetic query generation technique circumvents this problem by automatically generating artificial queries. For example, given the passage on the left, the synthetic query generator automatically generates a question similar to the one shown on the right.
+
+<img src="/assets/media/blog-images/2023-02-13-semantic-search-solutions/synthetic-query.png" alt="Synthetic query"/>{: .img-fluid }
+
+A medium-sized transformer model such as TAS-B can then be trained on several such `(query, passage)` pairs. Using this technique, we trained and released a large machine learning model that can create queries based on passages.
+
+**Performance**
+
+We conducted several tests to measure the performance of our technique. We generated synthetic queries for nine different challenge datasets and trained the TAS-B model on the dataset that contained the generated queries. When combined with BM25, our solution provides a 15% boost in search relevancy (measured in terms of [nDCG@10](https://en.wikipedia.org/wiki/Discounted_cumulative_gain)) over BM25. We found that the number of synthetic queries generated per passage drastically affects search performance, with more queries leading to better performance. Additionally, we found that the size of the synthetic query generator model also affects downstream performance: larger models lead to better synthetic queries, which in turn lead to better TAS-B models. 
+
+**Try it**
+
+To try the synthetic query generator model, follow the end-to-end guide in the [Demo Notebook for Sentence Transformer Model Training, Saving and Uploading to OpenSearch](https://opensearch-project.github.io/opensearch-py-ml/examples/demo_transformer_model_train_save_upload_to_openSearch.html). After you run this notebook, it will create a custom TAS-B model tuned to your corpus. You can then upload the model to the OpenSearch cluster and use it at query time by following the steps in the [Similar document search with OpenSearch](https://opensearch.org/blog/similar-document-search/#:~:text=%5D%0A%20%20%7D%0A%7D-,Search%20with%20k%2DNN,-In%20this%20example) blog.
```

Comment:
```suggestion
To try the synthetic query generator model, follow the end-to-end guide in the [Demo Notebook for Sentence Transformer Model Training, Saving and Uploading to OpenSearch](https://opensearch-project.github.io/opensearch-py-ml/examples/demo_transformer_model_train_save_upload_to_openSearch.html). After you run this notebook, it will create a custom TAS-B model tuned to your corpus. You can then upload the model to the OpenSearch cluster and use it at query time by following the steps in the [Similar document search with OpenSearch](https://opensearch.org/blog/similar-document-search/#:~:text=%5D%0A%20%20%7D%0A%7D-,Search%20with%20k%2DNN,-In%20this%20example) blog post.
```